apps/charterafrica/src/components/Comments/DynamicLineClampedTypography.js

```diff
@@ -0,0 +1,54 @@
+import { Button, useTheme } from "@mui/material";
+import PropTypes from "prop-types";
+import React, { forwardRef, useState, useRef } from "react";
+
+import LineClampedRichTypography from "@/charterafrica/components/LineClampedRichTypography";
+
+const DynamicLineClampedTypography = forwardRef(
+  ({ comment, ...props }, ref) => {
+    const ref1 = useRef();
+    const commentHeight = ref1?.current?.offsetHeight;
+    const currentTheme = useTheme();
+    const lineHeight =
+      currentTheme.typography.p3.lineHeight *
+      currentTheme.typography.p3.fontSize;
+    const showReadMore = commentHeight >= lineHeight * 3;
+
+    const [readMoreOpen, setReadMoreOpen] = useState(false);
```

Comment:
```suggestion
    const [showingMore, setShowingMore] = useState(false);
```
Or `open, setOpen`, etc.